user-management/build.gradle

```diff
@@ -0,0 +1,298 @@
+buildscript {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        maven { url "https://repo.spring.io/plugins-release" }
+    }
+    dependencies {
+        classpath "io.spring.gradle:propdeps-plugin:0.0.10.RELEASE"
+    }
+}
+
+plugins {
+    id "checkstyle"
+    id "java"
+    id "maven-publish"
+    id "idea"
+    id "jacoco"
+    id "org.springframework.boot"
+    id "com.google.cloud.tools.jib"
+    id "com.github.node-gradle.node"
+    id "com.diffplug.spotless"
+    // this allows us to find outdated dependencies via ./gradlew dependencyUpdates
+    id "com.github.ben-manes.versions"
+    id "com.github.andygoossens.gradle-modernizer-plugin"
+    id "org.openapi.generator" version "5.1.1" // version is included since the plugin is not used by Artemis
+    id "io.spring.nohttp" version "0.0.8" // version is included since the plugin is not used by Artemis
+    //jhipster-needle-gradle-plugins - JHipster will add additional gradle plugins here
+}
+
+group = "de.tum.in.www1.artemis"
+
+sourceCompatibility=16
+targetCompatibility=16
+
+apply from: "gradle/docker.gradle"
+apply from: "../gradle/liquibase.gradle"
+//jhipster-needle-gradle-apply-from - JHipster will add additional gradle scripts to be applied here
+
+if (project.hasProperty("prod")) {
+    apply from: "gradle/profile_prod.gradle"
+} else {
+    apply from: "gradle/profile_dev.gradle"
+}
+
+if (project.hasProperty("war")) {
+    apply from: "gradle/war.gradle"
+}
+
+if (project.hasProperty("zipkin")) {
+    apply from: "gradle/zipkin.gradle"
+}
+
+
+apply from: "../gradle/tasks.gradle"
+
+private excludedClassFilesForReport(classDirectories) {
```

Comment:
Isn't this already part of gradle.build?
As are the two ones below?

Reply:
I extracted this part because it will be different for each project. It defines the classes to include/exclude for the test coverage report. Since we are using source sets to share classes, the included classes from Artemis are also included in the report and the coverage drops because they are not tested here.
The excludedClassFilesForReport is the only part that is defined here + configuring it for the tasks jacocoTestReport and jacocoTestCoverageVerification. The rest is configured in tasks.gradle